terraform-provider-constellation/internal/provider/provider.go

```diff
@@ -42,33 +51,32 @@ func (p *ConstellationProvider) Metadata(_ context.Context, _ provider.MetadataR
 // Schema defines the HCL schema of the provider, i.e. what attributes it has and what they are used for.
 func (p *ConstellationProvider) Schema(_ context.Context, _ provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		Attributes: map[string]schema.Attribute{
-			"example_value": schema.StringAttribute{
-				MarkdownDescription: "Example provider attribute",
-				Optional:            true,
-			},
-		},
+		Description:         "The Constellation provider manages Constellation clusters.",
+		MarkdownDescription: `The Constellation provider manages Constellation clusters.`, // TODO(msanft): Provide a more sophisticated description.
 	}
 }
 
 // Configure is called when the provider block is initialized, and conventionally
 // used to setup any API clients or other resources required for the provider.
 func (p *ConstellationProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
 	// Populate the provider configuration model with what the user supplied when
-	// declaring the provider block.
+	// declaring the provider block. No-op for now, as no attributes are defined.
 	var data ConstellationProviderModel
 	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	// Configuration values are now available.
-	// if data.Endpoint.IsNull() { /* ... */ }
+	// Create the image-fetcher client.
+	imageFetcher := imagefetcher.New()
```

Comment:
since only the image data source should interact with this, I think that it should remain within the data source